pvlib/modelchain.py

```diff
@@ -877,19 +883,46 @@ def infer_spectral_model(self):
         params = _common_keys(module_parameters)
         if {'A4', 'A3', 'A2', 'A1', 'A0'} <= params:
             return self.sapm_spectral_loss
-        elif ((('Technology' in params or
-                'Material' in params) and
-               (self.system._infer_cell_type() is not None)) or
-              'first_solar_spectral_coefficients' in params):
+        elif "first_solar_spectral_coefficients" in params:
```

Comment:
Using the key "first_solar_spectral_coefficients" in `module_parameters` is not well-documented. A user would have to look at `PVSystem.first_solar_spectral_loss` method to know to do this. Not for this PR, a new issue I think.